WalletWasabi/Crypto/Secp256k1Helpers.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NBitcoin.Secp256k1
+{
+	public static class Secp256k1Helpers
+	{
+		public static bool Equals(GE a, GE b)
+		{
+			if (a.IsInfinity && b.IsInfinity)
+			{
+				return true;
+			}
+			else
+			{
+				return a.IsInfinity == b.IsInfinity && a.x == b.x && a.y == b.y;
```

Comment:
i can't think of a reason why side channel leaks mater for our use case WRT point comparison, but elsewhere nbitcoin uses bitwise AND to ensure there's no short circuiting in such comparisons

Reply:
side channel leaks are when a cryptographic implementation reveals information about secrets not through the APIs, for example by its power draw, by timing, or by error handling.

in this case `&&` will not evaluate statements on the right hand side if the left hand side is false, which might be a timing leak.

One such place is: https://github.com/MetacoSA/NBitcoin/blob/f1110703d79b1a88e67367d8d78e8aedf99d5d0a/NBitcoin/Secp256k1/ECPubKey.cs#L238-L239 but not that it doesn't take `IsInfinity` into account

Reply:
I found one. It's the same logic, but it's just a test helper for NBitcoin, so I'm not sure how rigorous.

```cs
private void ge_equals_ge(GE a, GE b)
{
	Assert.True(a.infinity == b.infinity);
	if (a.infinity)
	{
		return;
	}
	Assert.True(a.x == b.x);
	Assert.True(a.y == b.y);
}
```